File: Matlab/SCRoF_mars25/asGroupesDe.m
function gd=asGroupesDe(AS,brG,var)
% retourne les rangs des groupes des variables var dans AS.VG(brG).Gr
Gr=AS.VG(brG).Gr;
nv=numel(var);
gd=zeros(nv,1);
for k=1:nv
    for j=1:numel(Gr)
        if any(Gr{j}==var(k))
            break
        end
    end
    if any(Gr{j}==var(k))
        gd(k)=j;
    else
        error('La variable v%d n''est dans aucun groupe de AS.VG(%d)',var(k),brG)
    end
end

File: R0/asGroupesDe.R
asGroupesDe <- function(AS, brG, var) {
  # Retourne les rangs des groupes des variables var dans AS$VG[[brG]]$Gr
  
  Gr <- AS$VG[[brG]]$Gr
  nv <- length(var)
  gd <- numeric(nv)
  
  for (k in seq_len(nv)) {
    group_found <- FALSE
    
    for (j in seq_along(Gr)) {
      if (var[k] %in% Gr[[j]]) {
        group_found <- TRUE
        break
      }
    }
    
    if (group_found) {
      gd[k] <- j
    } else {
      stop(sprintf("La variable v%d n'est dans aucun groupe de AS$VG[%d]", var[k], brG))
    }
  }
  
  return(gd)
}